urwid/numedit.py

```diff
@@ -265,13 +265,9 @@ def value(self):
         Return the numeric value of self.edit_text.
         """
         if self.edit_text:
-            # get the ip and fp, handles also the case that there is no '.'
-            ip, fp = ([v for v in
-                      self.edit_text.split(self._decimalSeparator)] + [0])[0:2]
-
-            with localcontext() as ctx:
-                ctx = (self.significance or len(str(fp)))+1
-                return Decimal(self.edit_text.replace(
-                    self._decimalSeparator, '.')) * 1
+            normalized = Decimal(self.edit_text.replace(self._decimalSeparator, '.'))
+            if self.significance is not None:
+                return normalized.quantize(self.significance)
```

Comment:
https://docs.python.org/3/library/decimal.html#decimal.Decimal.quantize should be enough without context magic and manual calculation

Reply:
Not to mention I somehow wasn't using the context anyway...
And made the same mistake in the test case (facepalm).

This is a nice clean alternative.